### PR TITLE
automate release

### PR DESCRIPTION
**Pull request by ZhengzheYang**
automate the release for ASM 1.10

### CODE REVIEW

File: scripts/release-asm/common.sh [hunk 1]
```diff
@@ -1,4 +1,7 @@
 _DEBUG="${_DEBUG:=}"
+SCRIPT_PATH="${SCRIPT_PATH:=asmcli}"
+RELEASE_MODE="${RELEASE_MODE:=default}"
+PRECOMMIT="scripts/release-asm/precommit"
```
Comment (zerobfd) [inline scripts/release-asm/common.sh:4]: Looks like you never actually use this

File: scripts/release-asm/common.sh [hunk 2]
```diff
@@ -236,3 +243,112 @@ setup() {
...
+  fi
+}
+
+update_config_number() {
+  local OLD_NUMBER; OLD_NUMBER="${1}"
+  local NEW_NUMBER; NEW_NUMBER="${2}"
+
+  sed -i.bak "s/CONFIG_VER:=\"${OLD_NUMBER}\"/CONFIG_VER:=\"${NEW_NUMBER}\"/g" "${SCRIPT_PATH}.sh"
+
+  bazel build //:merge && cp ../bazel-bin/asmcli .
```
Comment (zerobfd) [inline scripts/release-asm/common.sh:272]: I think maybe you meant to use the pre-release here?

**Review by zerobfd — CHANGES_REQUESTED** [review]
All of the logic looks fine. It looks like none of it actually uses much stuff that's already in `common.sh`, so it feels like that's not the best place for it.

Maybe at first we should have a separate script for the "pre-release" steps, and then combine it with the existing release script/logic after we're comfortable with using it. What do you think?

File: scripts/release-asm/common.sh (hunk 2 above)
Comment (ZhengzheYang) [reply scripts/release-asm/common.sh:272]: the `precommit` script assumes it's at the package root so I have to re-write the steps here.

**Review by zerobfd — APPROVED** [review]

**Pull request merged**